awx/api/serializers.py

```diff
@@ -3884,15 +3884,24 @@ def get_stdout(self, obj):
         return UriCleaner.remove_sensitive(obj.stdout)
 
     def get_event_data(self, obj):
-        try:
-            return json.loads(
-                UriCleaner.remove_sensitive(
-                    json.dumps(obj.event_data)
+        # the project update playbook uses the git, hg, or svn modules
+        # to clone repositories, and those modules are prone to printing
+        # raw SCM URLs in their stdout (which *could* contain passwords)
+        # attempt to detect and filter HTTP basic auth passwords in the stdout
+        # of these types of events
+        task = obj.event_data.get('task_action') or {}
```

Comment:
defaulting to empty dict isn't making any sense to me here

Reply:
Yea, you're right.